lib/api/route/messages.dart

```diff
@@ -183,7 +183,7 @@ Future<SendMessageResult> sendMessage(
   final supportsReadBySender = connection.zulipFeatureLevel! >= 236; // TODO(server-8)
   return connection.post('sendMessage', SendMessageResult.fromJson, 'messages', {
     ...(switch (destination) {
-      StreamDestination() => {
+      ChannelDestination() => {
         'type': RawParameter('stream'),
```

Comment:
This is a similar story to #847 — StreamDestination describes part of our outgoing API where we say "stream", and so I'd like to package the rename together with starting to say "channel" in what we send. The server does accept "channel" here:
https://zulip.com/api/send-message#parameter-type